wireless-regdb: update to 2019-06-03

zacharyrs opened the pull request:
Fixes build issues on a python3 host (issues with the print statement
formatting in the current build)

Includes 100-regdb-write-firmware-file-format-version-code-20.patch and
other fixes.

Closes https://bugs.openwrt.org/index.php?do=details&task_id=1605.

Signed-off-by: Zachary Riedlshah <git@zacharyrs.me>

diizzyy commented:
You need python-future installed

diizzyy commented:
Use the URL instead of git

zacharyrs commented on package/firmware/wireless-regdb/Makefile:
```diff
@@ -1,12 +1,12 @@
+PKG_SOURCE:=wireless-regdb-master-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git/snapshot/
+PKG_HASH:=6737a742cbd4c9740171a537faff0b3ec2813632d74efb09bccfe39e605d7fb6
+PKG_CAT:=zcat
```
Probably not.

I added it as it was in the wiki example - this is the first time I've touched openwrt.

Removed it now.

diizzyy replied:
Really needed?

zacharyrs replied:
Apologies, is what really needed?

diizzyy replied:
```PKG_CAT```

zacharyrs commented:
Updated as requested, and squashed with previous commits.

zacharyrs commented:
I had to make a number of changes as I misunderstood how to format the Makefile.

The current version has a mix of spaces and tabs, so I swapped it all to spaces, which broke it. Now swapped back to tabs, which works.

Also had to define the PKG_BUILD_DIR, as the snapshot link includes 'master'.

Compiling fine locally now.

zacharyrs commented:
Apologies if you're busy, but I just thought I'd follow up regarding this...

robimarko commented:
@zacharyrs Thanks for the work, I was thinking about doing something similar but you beat me to it.
I also needed updated regdb

zacharyrs commented:
@robimarko Thanks, hopefully it'll get merged soon!

nasbdh9 commented:
If not run `pip install future` 
there will be an error `ImportError: No module named builtins`
Is this a problem with Python 2.x?

dengqf6 commented:
@diizzyy 
There are release tarballs in kernel.org mirror
`@KERNEL/software/network/wireless-regdb/wireless-regdb-2018.10.24.tar.xz`
Please use it instead

diizzyy commented:
If we have release tarballs by all means, go for it.

diizzyy commented:
@nasbdh9 
https://python-future.org/quickstart.html#installation

dengqf6 commented:
A newer version is available
`wireless-regdb-2019.03.01.tar.xz`

zacharyrs commented:
Sorry, never stuck this on my todo list so it sat there...
Looks like it builds fine!

Done now!

nbd168 commented:
Fails on my host (with python2) with the following error:

python /Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/db2fw.py /Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/regulatory.db /Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/db.txt
Traceback (most recent call last):
  File "/Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/db2fw.py", line 6, in <module>
    from dbparse import DBParser
  File "/Users/nbd/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.03.01/dbparse.py", line 3, in <module>
    from builtins import bytes
ImportError: No module named builtins

This will only build if the host has the "future" module installed, which we don't currently rely on.
I would like to avoid this extra host dependency if possible.

zacharyrs commented:
The future dependency is introduced in the 2018.05.09 version, which I think is when the scripts were migrated to be python 3 compatible.

I could create a couple of patches to try revert the changes that rely on futures, but that seems a little counterproductive, considering python 2 is only supported for another 9 months...

nbd168 commented:
Maybe you could make a host package that builds the module and installs it to staging dir, and then add some env vars or patches to make wireless-regdb use the module from there.

zacharyrs commented:
Unfortunately that sounds a bit out of my expertise...

I'm rather busy with uni and work right now, so this has a reasonably low priority.

I'll likely just manually patch the repo when I rebuild, or possibly create a custom feed, at least until python 3 is targeted.

diizzyy commented:
Having the OpenSSL bump in mind perhaps it's worth changing the dependency to python3 instead of doing it the last minute? https://pythonclock.org/

diizzyy commented:
Hmm....
Not sure if it's true for all installs but on my box (Alpine Linux) python2 packages get built before wireless-regdb which also adds a host version of python that doesn't have python-future resulting in a compilation failure.

orangepizza commented:
and now another new version https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git/tag/?h=master-2019-06-03

zacharyrs commented:
@orangepizza I'll update it this evening.

zacharyrs commented:
Sorted!

zacharyrs committed:
wireless-regdb: update to 2019.06.03

Fixes build issues on a python3 host (issues with the print statement
formatting in the current build).

Includes 100-regdb-write-firmware-file-format-version-code-20.patch and
other fixes.

Closes bugs.openwrt.org/index.php?do=details&task_id=1605.

Uses the tarball as requested.

Signed-off-by: Zachary Riedlshah <git@zacharyrs.me>

dengqf6 commented:
Why did you not use official release tarball anymore?

zacharyrs commented:
Woops! I just messed it up and grabbed an old commit.
Will swap it over now.

dengqf6 commented on package/firmware/wireless-regdb/Makefile:
```diff
@@ -1,22 +1,23 @@
+PKG_SOURCE_URL:=@KERNEL/software/network/wireless-regdb/
+PKG_HASH:=cd917ed86b63ce8d93947979f1f18948f03a4ac0ad89ec25227b36ac00dc54bf
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
`PKG_BUILD_DIR` can be removed. It's already the default

zacharyrs replied:
Fixed!

orangepizza commented:
newest patch fails to complie a image:

```
make[3]: Entering directory '/mnt/realdisk/openwrt/openwrt/package/firmware/wireless-regdb'
rm -f /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/.built
touch /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/.built_check
python /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/db2fw.py /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/regulatory.db /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/db.txt
Traceback (most recent call last):
  File "/mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/db2fw.py", line 6, in <module>
    from dbparse import DBParser
  File "/mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/dbparse.py", line 3, in <module>
    from builtins import bytes
ImportError: No module named builtins
make[3]: *** [Makefile:35: /mnt/realdisk/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/wireless-regdb-2019.06.03/.built] Error 1
make[3]: Leaving directory '/mnt/realdisk/openwrt/openwrt/package/firmware/wireless-regdb'
time: package/firmware/wireless-regdb/compile#0.11#0.03#0.13
make[2]: *** [package/Makefile:109: package/firmware/wireless-regdb/compile] Error 2
make[2]: Leaving directory '/mnt/realdisk/openwrt/openwrt'
make[1]: *** [package/Makefile:103: /mnt/realdisk/openwrt/openwrt/staging_dir/target-mipsel_24kc_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/mnt/realdisk/openwrt/openwrt'
make: *** [/mnt/realdisk/openwrt/openwrt/include/toplevel.mk:218: world] Error 2

```

orangepizza commented:
is there a way to force buildroot to use python3?

ynezz commented:
> is there a way to force buildroot to use python3?

Nope, see #1937

zacharyrs commented:
@orangepizza As per the comment above, you need `future` installed.

ynezz commented:
@zacharyrs Thanks a lot for your patience and continuous support! I've merged this into my staging tree, where I'm finalizing the Python3 support and I'm not able to build test it without this PR.

dengqf6 commented:
You can add `PKGARCH:=ALL` to `define Package/wireless-regdb` section, as it's an architecture-independent database file
